(강상문) Feature/solution database 리뷰 부탁드립니다.

kangsangmoon opened the pull request:
## #️⃣연관된 이슈

#

## 📝작업 내용

Example의 입력예제를 매개변수로 컴파일에 전달하여 컴파일 결과를 출력예제와 비교하여 일치하는지 판별

kangsangmoon committed:
refactor(Service) : Example update 삭제

kangsangmoon committed:
add(table) : Example 테이블 생성

kangsangmoon committed:
refactor(table) : Example 테이블과 연관관계 생성

kangsangmoon committed:
refactor(controller) : Example 업데이트 삭제

Duskafka commented on src/main/java/com/example/project/solution/domain/Example.java:
```diff
@@ -0,0 +1,37 @@
+        this.outExample = outExample;
+        this.setSolution(solution); // 관계 설정
+    }
+
+    public void setSolution(Solution solution) {
+        this.solution = solution;
+        if (!solution.getExamples().contains(this)) {
+            solution.addExample(this);
+        }
+    }
```
외부의 객체에 대한 정보가 있다고 해서 그 객체의 정보를 외부에서 조작하려고 하면 안 됩니다.
결합도에 대해서 공부해주세요

kangsangmoon committed:
add(Example) : 저장된 예제 변수 타입에 맞는 타입으로 변환

kangsangmoon committed:
update(response) : 리스폰스메시지 추가

kangsangmoon committed:
add(Example): Example repository

kangsangmoon committed:
update(Example): 예제 추가 로직 추가 및 변환 로직 추가

kangsangmoon committed:
update(Example): 예제 추가 로직 추가

kangsangmoon committed:
update(Example): 엔티티 클래스 수정

- @JoinColumn 삭제
- inExample 및 outExample 필드에 @Setter 어노테이션 추가
- setSolution 메서드 제거 및 생성자에서 직접 관계 설정

kangsangmoon committed:
update(yml): hibernate의 ddl-auto update로 수정

Duskafka committed:
fix(Exception): ExampleNotFindByIdException 추가

BusinessException을 하나 추가함.

Duskafka committed:
fix(Exception): ExampleNotFindByIdException 추가

BusinessException을 하나 추가함.

Duskafka committed:
update(Ranking): Ranking 기능

Ranking 기능을 만들기 위한 SQL문과 Repository Query문 작성

Duskafka committed:
fix(Enum): Pattern을 String으로 변경

복잡한 코드 사용을 줄이기 위하여 Pattern을 String으로 변경함

kangsangmoon committed:
Merge branch 'develop' into Feature/SolutionDatabase

kangsangmoon committed:
update(Solution): 테이블 이름 명시 및 @JoinColumn 어노테이션 제거

kangsangmoon committed:
feat(Example): ExampleRegisterRequest DTO 추가

- inExample, outExample, solutionId 필드 포함
- Solution 객체를 받아 Example 엔티티로 변환하는 toEntity 메서드 구현

kangsangmoon committed:
feat(Example): ExampleDeleteRequest DTO 추가

- solutionId와 exampleId 필드 포함

kangsangmoon committed:
update(Example): Example 관련 메서드 수정

- ExampleRegisterRequest 및 ExampleDeleteRequest DTO를 사용
- addExampleToSolution 및 removeExampleFromSolution 메서드 수정

kangsangmoon committed:
update(Example): Example 관련 요청 처리 로직 개선

- addExampleToSolution 메서드에서 request 객체에 solutionId 추가
- removeExampleFromSolution 메서드에 ExampleDeleteRequest 객체 생성 및 추가

kangsangmoon committed:
delete(Solution): Example 관련 메서드 삭제

kangsangmoon committed:
update(Compile): 예제를 넘기는 구문 추가

kangsangmoon committed:
update(Compile): 예제를 넘기는 구문 추가

kangsangmoon committed:
update(Compile): 예제 에러메시지 추가

kangsangmoon committed:
update(Compile): 예제 관련 코드 추가

kangsangmoon committed:
update(converter): 예제 변환 코드

kangsangmoon committed:
update(Compile): 컴파일 테스트 코드 추가

- 입력예제를 매개변수로 컴파일 서버에 전달
- 컴파일 결과와 출력예제가 일치하는지 확인하는 코드 추가

kangsangmoon committed:
update(Compile): 리스트가 아닌 예제ID로 예제를 받아온다

kangsangmoon committed:
update(Compile): 컴파일 서비스 역할별 분할 및 매개변수를 받아와 컴파일

kangsangmoon committed:
update(Example): getExampleById 메서드 추가 타입변환 메서드 제거

kangsangmoon committed:
update(Compile): 컴파일 서비스 역할 분할

kangsangmoon committed:
update(Compile): 컴파일 결과와 출력예제를 비교한 후 반환하는 response 메시지 추가

Duskafka commented on src/main/java/com/example/project/compile/dto/CompileRequest.java:
```diff
@@ -1,18 +1,10 @@
-    @Getter
-    private List<String> examples;
-
+    private Long exampleId;
```
exampleId가 아니라 Example의 input output을 보내야합니다

Duskafka commented on src/main/java/com/example/project/compile/service/CommandExecutorService.java:
```diff
@@ -0,0 +1,38 @@
+@Service
+public class CommandExecutorService {
+  //TODO 프로세스가 일정시간만 대기하도록 설정
+    public String runCommand(String command) throws IOException, InterruptedException {
```
예외를 던지면 이 예외는 어디서 처리해주나요

kangsangmoon committed:
delete(Compile): 사용하지 않는 클래스 삭제

kangsangmoon committed:
delete(Example): 사용하지 않는 클래스 삭제

kangsangmoon committed:
update(Example): 사용하지 않는 findAllBySolutionId 메서드 삭제

kangsangmoon committed:
update(Example): 예제 분할 메서드 추가 및 문제에 맞는 예제를 찾는 메서드 추가

kangsangmoon committed:
add(Example): 예제를 분할 하는 클래스 추가

kangsangmoon committed:
update(Example): 예제를 찾는 메서드 삭제

kangsangmoon committed:
update(gradle): gson 라이브러리를 추가하여 json 객체를 직렬화

kangsangmoon committed:
refactor(api): solutionId로 문제를 컴파일 할 수 있게 CompileApiController 수정

- compileCode 메서드를 solutionId를 경로 변수로 사용하도록 수정.
- 예제 입력값을 파싱하여 다중 입력을 처리하는 로직 추가.

kangsangmoon committed:
update(Compile): exampleId 제거

kangsangmoon committed:
refactor(Compile): CompileService를 HTTP 요청 기반 컴파일 방식으로 변경

- 기존 파일 기반 컴파일 방식을 HTTP 요청을 통해 컴파일 서버로 코드를 전송하는 방식으로 수정.
- 컴파일 서버와의 통신을 위한 HTTP 연결 및 JSON 페이로드 생성 로직 추가.
- 입력 파라미터와 출력 예제를 비교하여 결과를 반환하는 로직 구현.
- 컴파일 성공 및 실패에 대한 로깅 추가.

kangsangmoon committed:
refactor(Compile): static 제거
